controllers/discovery/initDiscoveryDomainData.js

```diff
@@ -14,6 +14,17 @@ const InitDiscoveryDomainData = async (req, res) => {
   const userId = req.userId;
 
   try {
+    let totalDataQuery = `SELECT 
+                      count(C.domain_page_id) as total_data
+                  FROM user_follow_domain A 
+                  INNER JOIN user_follow_domain B 
+                      ON A.domain_id_followed = B.domain_id_followed
+                      AND A.user_id_follower = :userId
+                  RIGHT JOIN domain_page C 
+                      ON C.domain_page_id = A.domain_id_followed
+                  WHERE C.credder_score >= ${CREDDER_MIN_SCORE} AND C.status = true
+                  `;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The SQL query `totalDataQuery` is vulnerable to SQL injection because the `CREDDER_MIN_SCORE` variable is directly concatenated into the query string. It's a good practice to use parameterized queries or prepared statements to prevent SQL injection attacks.
```diff
- WHERE C.credder_score >= ${CREDDER_MIN_SCORE} AND C.status = true
+ WHERE C.credder_score >= :credderMinScore AND C.status = true
```
Then, you can add `credderMinScore: CREDDER_MIN_SCORE` to the replacements object when executing the query.



<!-- This is an auto-generated comment by OSS CodeRabbit -->